JoinColony/block-ingestor
# Track extensions, colonies and actions from the latest processed block

This optimisation allows block-ingestor to pick up right where it left off before restart, rather than having to go from block 1. It uses `node-persist` library for easier persistent storage (which still uses JSON files, but has a nicer interface)

### Testing
I imagine this approach will only get tested property in the QA environment, but for now:
1. Run the ingestor on this branch separately
2. Run the `temp-create-data` script in CDapp
3. Restart the ingestor. On `master`, you'd see a bunch of failed mutations related to extensions and actions. On this branch, there shouldn't be more than 1-2 failed mutations (I suspect these are due to ganache/block-ingestor replaying some of the transactions)

TODO: 
- [x] Refactor stats to use `node-persist`
- [x] Track extensions from latest processed block
- [x] Track actions from latest processed block
